### PR TITLE
Issue #459 adding a new doc for fs.copyFile tests

### PR DESCRIPTION
**Pull request by YuechengWu**
added some tests for fs.copyFile() including flag tests.

### CODE REVIEW

**Comment by YuechengWu** [comment]
@CometS1 Hi Shawn, thank you for your suggestion. We did talk about it. Therefore, she did the test without the flag and I did the test with flag (sort of). We might collaborate later, but not sure at the moment.

File: tests/spec/fs.copyFile.spec.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  it('should be a function', function() {
+    var fs = util.fs();
+    expect(fs.copyFile).to.be.a('function');
+  });
+
+  it('shoud return an error if the source file does not exist', function(done) {
+    var fs = util.fs();
+
+    fs.copyFile('source.txt', 'destination.txt', function(error){
```
Comment (humphd) [inline tests/spec/fs.copyFile.spec.js]: Your paths need to be absolute in Filer: `/source.txt` and `/destination.txt`.

File: tests/spec/fs.copyFile.spec.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      expect(error).to.exist;
+      expect(error.code).to.equal('ENOENT');
+      done();
+    });
+  });
+
+  it('adding the flag as an argument, should return an error if the destination file already exist', function (done){
```
Comment (humphd) [inline tests/spec/fs.copyFile.spec.js]: exists (add an 's')

File: tests/spec/fs.copyFile.spec.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    var fs = util.fs();
+        
+    fs.copyFile('source.txt', 'destination.txt', COPYFILE_EXCL, function(error){
```
Comment (humphd) [inline tests/spec/fs.copyFile.spec.js]: Absolute paths here too.

File: tests/spec/fs.copyFile.spec.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+var util = require('../lib/test-utils.js');
+var expect = require('chai').expect;
+const fs = require('fs');
+const {COPYFILE_EXCL} = fs.constants;
+
+describe('fs.copyFile', function() {
```
Comment (humphd) [inline tests/spec/fs.copyFile.spec.js]: Let's add `.skip` to this, since we don't have an implementation yet.

https://mochajs.org/#inclusive-tests

**Comment by humphd** [comment]
Rebased and fixed, landed in https://github.com/filerjs/filer/commit/848cc7d3de336342a48b636a8d169ae9ff82aca0.

**Pull request closed without merging**